src/services/inplayer.account.service.ts

```diff
@@ -107,8 +107,8 @@ export const register: Register = async ({ config, email, password }) => {
 
 export const logout = async () => {
   try {
-    await InPlayer.Account.signOut();
-    return InPlayer.Notifications.unsubscribe();
+    InPlayer.Notifications.unsubscribe();
```

Comment:
We need await here. 
If somehow logout happens before the notifications.unsubscribe, the modal for simultaneous logins will show, which we don't want to in this case.

Reply:
I had the same thought, but

```typescript 
InPlayer.Notifications.unsubscribe()
```

returns void so awaiting it shouldn't make a difference in this case.

Reply:
![Uploading Screenshot 2023-06-06 at 15.53.11.png…]()

Reply:
<img width="568" alt="Screenshot 2023-06-06 at 15 53 11" src="https://github.com/jwplayer/ott-web-app/assets/104987342/7e3a46b6-6f58-4b8a-9ad2-0c5138c2a1a8">

Reply:
In that case we can maybe do smth like this:

await new Promise((resolve) => {
   InPlayer.Notifications.unsubscribe()
    resolve();
  });

Reply:
There is no need to wrap the unsubscribe call with a promise when it doesn't return a promise.

When the function is blocking, it should have unsubscribed before proceeding to the `signOut` call.

This is essentially the same as your suggestion:

```ts
InPlayer.Notifications.unsubscribe();
InPlayer.Accounts.signOut();
```

In case the unsubscribe method is not blocking, a different method needs to be implemented to acknowledge the subscriptions are disposed before calling `signOut`, but I assume this is not the case.

Reply:
Yeah, you are right. The promise will resolve instantly after calling the unsubscribe and we have pretty much the same effect.
Then I guess it's okay to stay like this, but I am afraid if somehow unsubscribe does not happen and logout is made, the simultaneous login modal will popup..